### ISSUE TITLE
Merging results doesn't propagate the same exit status

### ISSUE THREAD

**Issue by jpeach** [dominikh/go-tools#1479]
- The output of 'staticcheck -version'
```
$ staticcheck -version
staticcheck 2023.1.6 (v0.4.6)
```
- The output of 'staticcheck -debug.version' (it is fine if this command fails)
```
$ staticcheck -debug.version
staticcheck 2023.1.6 (v0.4.6)

Compiled with Go version: go1.21.1
Main module:
	honnef.co/go/tools@v0.4.6 (sum: h1:oFEHCKeID7to/3autwsWfnuv69j3NsfcXbvJKuIcep8=)
Dependencies:
	github.com/BurntSushi/toml@v1.2.1 (sum: h1:9F2/+DoOYIOksmaJFPw1tGFy1eDnIJXg+UHjuD8lTak=)
	golang.org/x/exp/typeparams@v0.0.0-20221208152030-732eee02a75a (sum: h1:Jw5wfR+h9mnIYH+OtGT2im5wV1YGGDora5vTv/aa5bE=)
	golang.org/x/mod@v0.12.0 (sum: h1:rmsUpXtvNzj340zd98LZ4KntptpfRHwpFOHG188oHXc=)
	golang.org/x/sys@v0.11.0 (sum: h1:eG7RXZHdqOJ1i+0lgLgCpSXAp6M3LYlAo6osgSi0xOM=)
	golang.org/x/tools@v0.12.1-0.20230825192346-2191a27a6dc5 (sum: h1:Vk4mysSz+GqQK2eqgWbo4zEO89wkeAjJiFIr9bpqa8k=)
```
- The output of 'go version'
```
$ go version
go version go1.21.1 linux/amd64
```
- The output of 'go env'
```
$ go env
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/home/docker/.cache/go-build'
GOENV='/home/docker/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/src/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/src/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.21.1'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/Users/James-Peach/.../go.mod'
GOWORK='/Users/James-Peach/.../go.work'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build1307489963=/tmp/go-build -gno-record-gcc-switches'
```

I have a project that follows the documentation about merging results for build tags. It appears that if you do this, the merge process can lose the previous exit status. This means that in CI, the staticcheck job succeeds when it should fail.

Running staticcheck:
```
$ staticcheck ./...
..._test.go:4:2: "io/ioutil" has been deprecated since Go 1.19: As of Go 1.16, the same functionality is now provided by package [io] or package [os], and those implementations should be preferred in new code. See the specific function documentation for details.  (SA1019)
.../main.go:8:2: "io/ioutil" has been deprecated since Go 1.19: As of Go 1.16, the same functionality is now provided by package [io] or package [os], and those implementations should be preferred in new code. See the specific function documentation for details.  (SA1019)
...
$ echo $?
1
```

Running it with merge:
```
$ staticcheck -f binary ./... | staticcheck -merge
... elided the same set of issues ...
$ echo $?
0
```

**Comment by dominikh** [comment]
Indeed. This is fixed in 15308ddc45fd76ad1f23b3aae5ee7f8e645981c7. As a workaround in the meantime, if you don't want to update to an unreleased version, you can check if anything was written to stdout and fix the exit status.

**Comment by jpeach** [comment]
> Indeed. This is fixed in [15308dd](https://github.com/dominikh/go-tools/commit/15308ddc45fd76ad1f23b3aae5ee7f8e645981c7). As a workaround in the meantime, if you don't want to update to an unreleased version, you can check if anything was written to stdout and fix the exit status.

Great! Thanks a lot! 👍
